src/test/java/io/jenkins/plugins/analysis/core/history/PreviousRunReferenceTest.java

```diff
@@ -0,0 +1,522 @@
+        Run baseline = mock(Run.class);
+        Run previousRunMock = mock(Run.class);
+        when(baseline.getPreviousBuild()).thenReturn(previousRunMock);
+        when(previousRunMock.getResult()).thenReturn(Result.UNSTABLE);
+
+        ResultAction action = mock(ResultAction.class);
+        when(action.isSuccessful()).thenReturn(true);
+
+        AnalysisResult analysis = mock(AnalysisResult.class);
+        when(action.getResult()).thenReturn(analysis);
+        when(analysis.getOverallResult()).thenReturn(Result.UNSTABLE);
+
+        ResultSelector selector = mock(ResultSelector.class);
+        when(selector.get(previousRunMock)).thenReturn(Optional.of(action));
+
+        List<ResultAction> actions = Collections.singletonList(action);
+        when(previousRunMock.getActions(ResultAction.class)).thenReturn(actions);
+
+        PreviousRunReference previousRun = new PreviousRunReference(baseline, selector,
+                false);
+
+        Optional<ResultAction> resultAction = previousRun.getReferenceAction();
+
+        verify(selector, times(2)).get(previousRunMock);
+        assertThat(resultAction).isEqualTo(Optional.of(action));
+    }
+
+    /**
+     * Test that there is one unstable previous action, with overallResultMusBeSuccess = true.
+     */
+    @Test
+    void oneUnstableAction_withSuccessful_shouldReturnNothing() {
+        Run baseline = mock(Run.class);
+        Run previousRunMock = mock(Run.class);
+        when(baseline.getPreviousBuild()).thenReturn(previousRunMock);
+        when(previousRunMock.getResult()).thenReturn(Result.UNSTABLE);
+
+        ResultAction action = mock(ResultAction.class);
+        when(action.isSuccessful()).thenReturn(true);
+
+        AnalysisResult analysis = mock(AnalysisResult.class);
+        when(action.getResult()).thenReturn(analysis);
+        when(analysis.getOverallResult()).thenReturn(Result.UNSTABLE);
+
+        ResultSelector selector = mock(ResultSelector.class);
+        when(selector.get(previousRunMock)).thenReturn(Optional.of(action));
+
+        List<ResultAction> actions = Collections.singletonList(action);
+        when(previousRunMock.getActions(ResultAction.class)).thenReturn(actions);
+
+        PreviousRunReference previousRun = new PreviousRunReference(baseline, selector,
+                true);
+
+        Optional<ResultAction> resultAction = previousRun.getReferenceAction();
+
+        verify(selector, times(1)).get(previousRunMock);
+        assertThat(resultAction).isEmpty();
+    }
+
+    /**
+     * Test that there are two succesful previous action, with overallResultMusBeSuccess = false.
+     */
+    @Test
+    void twoTrueAction_withAnyResult_shouldReturnTwo() {
+        Run baseline = mock(Run.class);
+        Run previousRunMock = mock(Run.class);
+        when(baseline.getPreviousBuild()).thenReturn(previousRunMock);
+        when(previousRunMock.getResult()).thenReturn(Result.SUCCESS, Result.SUCCESS);
```

Comment:
it's the same as just
when(previousRunMock.getResult()).thenReturn(Result.SUCCESS);